cmd/object-handlers.go

```diff
@@ -287,6 +287,9 @@ func (api objectAPIHandlers) HeadObjectHandler(w http.ResponseWriter, r *http.Re
 // Extract metadata relevant for an CopyObject operation based on conditional
 // header values specified in X-Amz-Metadata-Directive.
 func getCpObjMetadataFromHeader(header http.Header, defaultMeta map[string]string) (map[string]string, error) {
+	// Make sure to remove saved etag if any, CopyObject calculates a new one.
+	delete(defaultMeta, "etag")
```

Comment:
passed `defaultMeta` is a pointer.  Do we want to modify the source?

Reply:
Yes we need to remove "etag" we can make it safer.

Reply:
what do you mean by safer?  what is the problem with passed ETag?

Reply:
you shouldn't pass the "etag" in CopyObject @balamurugana